File: day1/task2.erl
-module(task2).
-export([main/0]).

-spec basement(binary(), integer(), integer()) -> integer().
basement(<<"(", Tail/binary>>, Floor, Position) ->
  basement(Tail, Floor + 1, Position + 1);
basement(<<")", Tail/binary>>, Floor, Position) 
  when Floor > 0 -> basement(Tail, Floor - 1, Position + 1);
basement(<<")", _/binary>>, Floor, Position) 
  when Floor =:= 0 -> Position.

-spec main() -> none().
main() -> 
  {ok, Binary} = file:read_file("input.txt"),
  Floor = basement(Binary, 0, 1),
  io:format("The position is ~p.\n", [Floor]).

File: day1/task2.jl
function entrance(stream :: IOStream) :: Int32
  floor :: Int32 = 0
  position :: Int32 = 1
  while !eof(stream)
    symbol :: Char = read(stream, Char)
    if symbol == '('
      floor += 1
    elseif symbol == ')'
      floor -= 1
    end
    if floor < 0 
      return position 
    end
    position += 1
  end
  return -1
end

open("input.txt") do file
  answer :: Int32 = entrance(file)
  println(answer)
end
